p2p/pb/proto_services.go

```diff
@@ -134,7 +134,10 @@ func EncodeQuaiResponse(id uint32, location common.Location, data interface{}) (
 		respMsg.Response = &QuaiResponseMessage_Transaction{Transaction: protoTransaction}
 
 	case *trie.TrieNodeResponse:
-		protoTrieNode := &trie.ProtoTrieNode{ProtoNodeData: data.NodeData}
+		protoTrieNode := &trie.ProtoTrieNode{
+			ProtoNodeData: data.NodeData,
+			ProtoNodeHash: data.NodeHash.Bytes(),
```

Comment:
Please do not include the node hash in the encoded node. The hash should be the hash of the encoding, which cannot include itself